bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/discovery/AstroDiscoveryService.java

```diff
@@ -44,41 +43,29 @@
  * @author Gerhard Riegler - Initial Contribution
  * @author Stefan Triller - Use configured location
  */
+@NonNullByDefault
 @Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.astro")
 public class AstroDiscoveryService extends AbstractDiscoveryService {
-    private final Logger logger = LoggerFactory.getLogger(AstroDiscoveryService.class);
     private static final int DISCOVER_TIMEOUT_SECONDS = 2;
     private static final int LOCATION_CHANGED_CHECK_INTERVAL = 60;
-    private LocationProvider locationProvider;
-    private ScheduledFuture<?> astroDiscoveryJob;
-    private PointType previousLocation;
 
     private static final ThingUID SUN_THING = new ThingUID(THING_TYPE_SUN, LOCAL);
     private static final ThingUID MOON_THING = new ThingUID(THING_TYPE_MOON, LOCAL);
 
-    /**
-     * Creates a AstroDiscoveryService with enabled autostart.
-     */
-    public AstroDiscoveryService() {
-        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, true);
-    }
+    private final Logger logger = LoggerFactory.getLogger(AstroDiscoveryService.class);
 
-    @Override
-    @Activate
-    protected void activate(Map<String, Object> configProperties) {
-        super.activate(configProperties);
-    }
+    private final LocationProvider locationProvider;
 
-    @Override
-    @Modified
-    protected void modified(Map<String, Object> configProperties) {
-        super.modified(configProperties);
-    }
+    private @Nullable ScheduledFuture<?> astroDiscoveryJob;
+    private @Nullable PointType previousLocation;
 
-    @Override
-    @Deactivate
-    protected void deactivate() {
-        super.deactivate();
+    @Activate
+    public AstroDiscoveryService(final @Reference LocationProvider locationProvider,
+            final @Reference LocaleProvider localeProvider, final @Reference TranslationProvider i18nProvider) {
+        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, true);
+        this.locationProvider = locationProvider;
+        this.localeProvider = localeProvider;
+        this.i18nProvider = i18nProvider;
```

Comment:
You need to call activate otherwise background discovery will not start.

```suggestion
            final @Reference LocaleProvider localeProvider, final @Reference TranslationProvider i18nProvider,
            Map<String, Object> configProperties) {
        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, true);
        this.locationProvider = locationProvider;
        this.localeProvider = localeProvider;
        this.i18nProvider = i18nProvider;
        activate(configProperties);
```